core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,344 @@
+    spyOn(statesToQuestionsService, 'getQuestions').andReturn(null);
+    expectedReturnedValue = simpleEditorManagerService.tryToInit();
+    expect(expectedReturnedValue).toBeFalsy();
+  });
+
+  it('should return data', function() {
+    // Expected initialized data inside simpleEditorManagerService
+    var expectedSimpleEditorManagerData = {
+      title: data.title,
+      introductionHtml: data.states.Introduction.content[0].value,
+      questionList: null
+    };
+    // Initialize simpleEditorManagerService.data
+    simpleEditorManagerService.tryToInit();
+    expectedSimpleEditorManagerData.questionList = questionList;
+    expect(simpleEditorManagerService.getData())
+      .toEqual(expectedSimpleEditorManagerData);
+  });
+
+  it('should return title', function() {
+    var expectedTitle = data.title;
+    // Initialize simpleEditorManagerService.data
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getTitle()).toEqual(expectedTitle);
+  });
+
+  it('should return introduction html', function() {
+    var expectedIntroductionHtml = data.states.Introduction.content[0].value;
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toEqual(expectedIntroductionHtml);
+  });
+
+  it('should return question list', function() {
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getQuestionList())
+      .toEqual(expectedSimpleEditorManagerQuestionList);
+  });
+
+  it('should save the given new title', function() {
+    var expectedNewTitle = 'ExpNewTitle';
+    simpleEditorManagerService.saveTitle(expectedNewTitle);
+    expect(explorationTitleService.savedMemento).toEqual(expectedNewTitle);
+    expect(simpleEditorManagerService.getTitle()).toEqual(expectedNewTitle);
+  });
+
+  it('should save the given new introduction html', function() {
+    var expectedNewIntroductionHtmll = '<p> new intro </p>';
+    simpleEditorManagerService
+      .saveIntroductionHtml(expectedNewIntroductionHtmll);
+    expect(simpleEditorShimService.getIntroductionHtml())
+      .toEqual(expectedNewIntroductionHtmll);
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toEqual(expectedNewIntroductionHtmll);
+  });
+
+  it('should save the customization args', function() {
+    var testStateName = 'Introduction';
+    var expectedNewCustomizationArgs = {
+      choices: {
+        value: [
+          '<p>Option 3</p>'
+        ]
+      }
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveCustomizationArgs(
+      testStateName, expectedNewCustomizationArgs);
+    expect(explorationStatesService.getState(testStateName)
+      .interaction.customization_args).toEqual(expectedNewCustomizationArgs);
+  });
+  /**
+    * Not working, needs debugging.
+    It('should save the answer groups', function () {
```

Comment:
In develop, that function does exist: https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js#L28

It also exists in simple-editor: https://github.com/oppia/oppia/blob/simple-editor/core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js

Perhaps you've discovered a bug. Can you check to make sure that answer group domain objects are actually AnswerGroup objects (and not dicts)?

Also, it might be worth merging simple-editor into your branch. There have been some updates to this part of the code recently (by @tjiang11) that might be related to this, so it's worth ensuring that you're in sync with the latest code.